api_app/analyzers_manager/observable_analyzers/tweetfeeds.py

```diff
@@ -0,0 +1,123 @@
+import json
+import logging
+import os
+from typing import Tuple
+
+import requests
+from django.conf import settings
+
+from api_app.analyzers_manager.classes import ObservableAnalyzer
+from api_app.analyzers_manager.exceptions import AnalyzerRunException
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class TweetFeeds(ObservableAnalyzer):
+    """
+    wrapper for https://tweetfeed.live
+    """
+
+    url = "https://api.tweetfeed.live/v1/"
+    filter1: str = ""
+    update_on_run: bool = True
+    time: str = "test_time"  # for tests
+    observable_classification: str = "test"  # for tests
```

Comment:
mmmm test things should remain in tests, cannot be the default value of these attributes.

Considering that the `update` works hourly by default, we need to put here decent values and we need also to add those as default parameters in the migration.

`observable_classification` should not exist while `time` should be a default, Idk, a week, a month?

Reply:
Okay so I have some doubts here, Example if I did a run with some parameters that generate a unique url to download, I would want my update function to update the db based on that generated url every hour or so. Here the update function wrt the framework in place requires me to hardcode the url for update which seems unusable/irrelevant in this use case, where the url can be unique/new/user based

Reply:
Hmm, okay, so you are saying to have one generic database url hardcoded. If the user wants they can extend the default database by using update on run parameter and providing other parameters at the same time. We would require other url parameters only and only if the update on run is true. Else in every run, we can simply look up the default db.
It does make sense and simplifies things a lot and can be implemented.

However, in the longer run, I do believe that the update function should have some mechanism to be able to cache or sustain the values of the parameters used in the last run of such analyzers, this could save us extra computation by downloading and updating the exact database required by the user instead of relying on having a larger superset of the the database that the user **might** use.
Example:
In our case as you suggest, we are downloading data for all the supported observables types for a month and updating it every hour. If I'm a user who is only looking for the data from a particular user with time set to just `today` with classification type `url` why would I want my software to download, update and search for data of past 30 days that has entries of all other observables including `url`.

Reply:
I understand but consider that the problem is a lot bigger than what you think because if you need to manage the data by the user you should also start to leverage and check all things related to permissions, organizations and access and things can become quite messy very fast.

You would need to create a different cache for every user based on each user custom parameters and manage application and file access.

Again, I'd stick for the simple one. It benefits maintenance and stability.

> In our case as you suggest, we are downloading data for all the supported observables types for a month and updating it every hour.

not every hour, it is too much, we can do that once in a day.

> I would want to cache and update only for the database that my generated url gives and not care about a superset of my use case db

yeah you could add another boolean option to manage the case where the user does use only the data downloaded at the time of request but again, simple is better. Consider that the normal user does not know all the details that we are talking about. He needs to have something simple to understand and that works. That's all

Reply:
Understood! Let's move ahead with what you proposed. I'm travelling right now, so I will probably commit the updates tomorrow.

Reply:
np man, take your time :)